driver/search_grpc.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"unsafe"
+
+	api "github.com/tigrisdata/tigris-client-go/api/server/v1"
+)
+
+type grpcSearch struct {
+	Project string
+
+	search api.SearchClient
+}
+
+func NewGRPCSearchClient(project string, client api.SearchClient) SearchClient {
+	return &grpcSearch{Project: project, search: client}
+}
+
+func (c *grpcSearch) CreateOrUpdateIndex(ctx context.Context, name string, schema json.RawMessage) error {
+	_, err := c.search.CreateOrUpdateIndex(ctx, &api.CreateOrUpdateIndexRequest{
+		Project: c.Project,
+		Name:    name,
+		Schema:  schema,
+	})
+
+	return err
+}
+
+func (c *grpcSearch) GetIndex(ctx context.Context, name string) (*IndexInfo, error) {
+	resp, err := c.search.GetIndex(ctx, &api.GetIndexRequest{
+		Project: c.Project,
+		Name:    name,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &IndexInfo{Name: resp.Index.Name, Schema: resp.Index.Schema}, nil
+}
+
+func (c *grpcSearch) DeleteIndex(ctx context.Context, name string) error {
+	_, err := c.search.DeleteIndex(ctx, &api.DeleteIndexRequest{
+		Project: c.Project,
+		Name:    name,
+	})
+
+	return err
+}
+
+func (c *grpcSearch) ListIndexes(ctx context.Context, filter *IndexSource) ([]*IndexInfo, error) {
+	if filter == nil {
+		filter = &IndexSource{}
+	}
+
+	resp, err := c.search.ListIndexes(ctx, &api.ListIndexesRequest{
+		Project: c.Project,
+		Filter:  filter,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetIndexes(), nil
+}
+
+func (c *grpcSearch) Get(ctx context.Context, name string, ids []string) ([]*IndexDoc, error) {
+	resp, err := c.search.Get(ctx, &api.GetDocumentRequest{
+		Project: c.Project,
+		Index:   name,
+		Ids:     ids,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetDocuments(), nil
+}
+
+func (c *grpcSearch) CreateByID(ctx context.Context, name string, id string, doc json.RawMessage) error {
+	_, err := c.search.CreateById(ctx, &api.CreateByIdRequest{
+		Project:  c.Project,
+		Index:    name,
+		Id:       id,
+		Document: doc,
+	})
+
+	return err
+}
+
+func (c *grpcSearch) Create(ctx context.Context, name string, docs []json.RawMessage) error {
+	_, err := c.search.Create(ctx, &api.CreateDocumentRequest{
```

Comment:
Create also returns ids, we need to expose it because id can be missing in a document and will be auto-generated by the server.